Cgen error for multi return functions

### Describe the bug

See code: https://play.vlang.io/p/91d272c370

```v
fn goo() (string, int) {
	return "", 100
}

fn foo() (string, int, bool) {
	return goo(), true
}

fn main() {
	println(foo())
}
```

### Expected Behavior

No cgen error

### Current Behavior

```
/tmp/v_501/main.16715658530030987975.tmp.c:12278:31: error: initializing 'multi_return_int_string_bool' (aka 'struct multi_return_int_string_bool') with an expression of incompatible type 'multi_return_int_string' (aka 'struct multi_return_int_string')
        multi_return_int_string_bool _t1 = main__goo();
builder error                                     ^     ~~~~~~~~~~~
```

### Reproduction Steps

Run code above

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

0.3.3

### Environment details (OS name and version, etc.)

Playground

--- a/vlib/v/checker/return.v
+++ b/vlib/v/checker/return.v
@@ -82,6 +82,9 @@ fn (mut c Checker) return_stmt(mut node ast.Return) {
 		// Unpack multi return types
 		sym := c.table.sym(typ)
 		if sym.kind == .multi_return {
+			if i > 0 || i != node.exprs.len - 1 {
+				c.error('cannot use multi-return with other return types', expr.pos())
+			}
 			for t in sym.mr_info().types {
 				got_types << t
 				expr_idxs << i

--- a/vlib/v/checker/tests/multireturn_mix_err.out
+++ b/vlib/v/checker/tests/multireturn_mix_err.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/multireturn_mix_err.vv:6:9: error: cannot use multi-return with other return types
+    4 | 
+    5 | fn foo() (string, int, bool) {
+    6 |     return goo(), true
+      |            ~~~~~
+    7 | }

--- a/vlib/v/checker/tests/multireturn_mix_err.vv
+++ b/vlib/v/checker/tests/multireturn_mix_err.vv
@@ -1,0 +1,7 @@
+fn goo() (string, int) {
+	return '', 100
+}
+
+fn foo() (string, int, bool) {
+	return goo(), true
+}

